ros2_foxglove_bridge_sdk/src/ros2_foxglove_bridge.cpp

```diff
@@ -382,32 +383,31 @@ void FoxgloveBridge::updateAdvertisedServices() {
   std::lock_guard<std::mutex> lock(_servicesMutex);
 
   // Remove advertisements for services that have been removed
-  std::vector<foxglove_ws::ServiceId> servicesToRemove;
-  for (const auto& service : _advertisedServices) {
-    const auto it = std::find_if(serviceNamesAndTypes.begin(), serviceNamesAndTypes.end(),
-                                 [service](const auto& serviceNameAndTypes) {
-                                   return serviceNameAndTypes.first == service.second.name;
-                                 });
-    if (it == serviceNamesAndTypes.end()) {
-      servicesToRemove.push_back(service.first);
+  std::vector<std::string> servicesToRemove;
+  for (const auto& [serviceName, _] : _advertisedServices) {
+    if (serviceNamesAndTypes.find(serviceName) == serviceNamesAndTypes.end()) {
+      servicesToRemove.push_back(serviceName);
     }
   }
-  for (auto serviceId : servicesToRemove) {
-    _advertisedServices.erase(serviceId);
+  for (auto serviceName : servicesToRemove) {
```

Comment:
```suggestion
  for (const auto& serviceName : servicesToRemove) {
```

otherwise you are making a copy of every string